### PR TITLE
Allow joining open conversations with passwords

### PR DESCRIPTION
**Pull request by nickvergessen**
See integration test for steps

### CODE REVIEW

File: tests/integration/features/conversation/join-listable.feature [hunk 1]
```diff
@@ -95,3 +95,18 @@ Feature: conversation/join-listable
...
+  # -----------------------------------------------------------------------------
+  # Join listed conversation which has a password
+  # -----------------------------------------------------------------------------
+  Scenario: Only users with accounts can join an all-listed group room
+    Given user "creator" creates room "room" (v4)
+      | roomType | 3    |
+      | roomName | room |
+    When user "creator" sets password "foobar" for room "room" with 200 (v4)
+    And user "creator" allows listing room "room" for "all" with 200 (v4)
+    When user "regular-user" joins room "room" with 200 (v4)
```
Comment (vitormattos) [inline tests/integration/features/conversation/join-listable.feature:108]: Password isn't mandatory to join room?

Comment (nickvergessen) [reply tests/integration/features/conversation/join-listable.feature:108]: So if it's not listable that makes totally sense to be mandatory.
But when it's listable and you find it via search having to enter the password sounds like a very awkward flow.

**Review by vitormattos — APPROVED** [review]

**Pull request merged**